common/metrics/defs.go

```diff
@@ -1546,6 +1554,12 @@ var MetricDefs = map[ServiceIdx]map[int]metricDefinition{
 		ArchiverHandleAllRequestsLatency:         {metricName: "archiver_handle_all_requests_latency", oldMetricName: "archiver.handle-all-requests-latency"},
 		ArchiverWorkflowStoppingCount:            {metricName: "archiver_workflow_stopping", oldMetricName: "archiver.workflow-stopping"},
 		ArchiverClientSendSignalFailureCount:     {metricName: "archiver_client_send_signal_error", oldMetricName: "archiver.client-send-signal-error"},
+		TaskProcessedCount:                       {metricName: "task.processed", metricType: Gauge},
```

Comment:
names should have `_` in them . not allowed.. 

Adding a test that will catch this. https://github.com/uber/cadence/pull/1603

Reply:
Fixed !